pkg/logger/progress_logger.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2022 VMware, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logger
+
+import (
+	"context"
+	"fmt"
+
+	pb "github.com/cheggaaa/pb/v3"
+	regv1 "github.com/google/go-containerregistry/pkg/v1"
+)
+
+// ProgressLogger interface should match with imgpkg util Progress logger
+type ProgressLogger interface {
+	Start(ctx context.Context, progress <-chan regv1.Update)
+	End()
+}
+
+// NewProgressBar constructor to build a ProgressLogger responsible for printing out a progress bar using updates when
+// writing to a registry via ggcr
+func NewProgressBar() ProgressLogger {
+	return &progressBar{}
+}
+
+// ProgressBarLogger display progress bar on output
+type progressBar struct {
+	cancelFunc context.CancelFunc
+	bar        *pb.ProgressBar
+}
+
+// Start the display of the Progress Bar
+func (l *progressBar) Start(ctx context.Context, progressChan <-chan regv1.Update) {
+	ctx, cancelFunc := context.WithCancel(ctx)
+	l.cancelFunc = cancelFunc
+	// Add a new empty line to separate the progress bar from prior output
+	fmt.Println()
+	l.bar = pb.New64(0)
+	l.bar.Set(pb.Bytes, true)
+	l.bar.Set(pb.SIBytesPrefix, true)
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case update := <-progressChan:
+				if update.Error != nil {
+					continue
+				}
+				if update.Total == 0 {
+					return
+				}
+				if !l.bar.IsStarted() {
+					l.bar.SetTotal(update.Total)
+					l.bar.Start()
+				}
+				l.bar.SetCurrent(update.Complete)
+				l.bar.Write()
+			}
+		}
+	}()
+}
+
+// End stops the progress bar and writes the final message
```

Comment:
Does `l.cancelFunc()` or `l.bar.Finish()` update writes a final message? if not update the comment